extensions/control-plane/api/management-api/contract-negotiation-api/src/main/java/org/eclipse/edc/connector/api/management/contractnegotiation/ContractNegotiationNewApiController.java

```diff
@@ -74,8 +73,7 @@ public List<JsonObject> queryNegotiations(JsonObject querySpecDto) {
 
         Function<Result<JsonObject>, Result<QuerySpec>> expandedMapper =
                 expandedResult -> expandedResult
-                        .compose(jsonObject -> transformerRegistry.transform(jsonObject, QuerySpecDto.class))
-                        .compose(dto -> transformerRegistry.transform(dto, QuerySpec.class));
+                        .compose(jsonObject -> transformerRegistry.transform(jsonObject, QuerySpec.class));
```

Comment:
Why we're avoiding to pass by the Dto? This is done also in the other `*NewApiController` as policy def, contract def, ... we need either to do it everywhere or nowhere, I'd choose the latter, mostly for consistency.

Reply:
Because now we have `JsonObject <-> QuerySpectDto` but also i introduced  `JsonObject <-> QuerySpec` for the protocol stuff which basically are the same. So i though we can just remove the passing through a Dto since there is a direct ``JsonObject <-> QuerySpec` transformation. But if we need that for consistency i can revert back to DTO.
In case i would distinguish the QuerySpec from QuerySpecDto from the type prospective

Reply:
I mean, I could agree, but in case we'd need to replace it everywhere, not only for CN and TP.
I'd leave this refactoring (with a committer agreement before) to a subsequent issue/pr

Reply:
Or if we want to remove this indirection later probably it make sense to collapse the types

Reply:
What do you think we leave the QuerySpecDto and CriterionDto in place but we name the type as `QuerySpec` and `Criterion` ?

Reply:
in any event it should be consistent or all *Dto or not :)

Reply:
@ndr-brt  i've restored the DTOs, and also changed `QuerySpecDto` to be `QuerySpecDto` that will be consistent with `CriterionDto`